.github/workflows/kfp-kubernetes-execution-tests.yml

```diff
@@ -73,7 +73,25 @@ jobs:
         run: pip install -r ./test/kfp-kubernetes-execution-tests/requirements.txt
 
       - name: Run tests
+        id: test
         run: |
           export KFP_ENDPOINT="http://localhost:8888"
           export TIMEOUT_SECONDS=2700
           pytest ./test/kfp-kubernetes-execution-tests/sdk_execution_tests.py --asyncio-task-timeout $TIMEOUT_SECONDS
+        continue-on-error: true
+
+      - name: Collect failed logs
+        if: steps.test.outcome != 'success'
+        run: ./.github/resources/scripts/collect-logs.sh --ns kubeflow --output /tmp/tmp_pod_log.txt
+        continue-on-error: true
+
+      - name: Collect test results
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: kfp-execution-tests-artifacts-k8s-${{ matrix.k8s_version }}
+          path: /tmp/tmp*/*
+          
+      - name: Fail test
+        if: steps.test.outcome != 'success'
+        run: exit 1
```

Comment:
You don't need this step. You can just `exit 1` in `Collect failed logs` like done [here](https://github.com/opendatahub-io/data-science-pipelines-operator/pull/747/files#diff-6b1da844f6607892239868957a7d5c6a76e10c007f623ba88e89d30b6e4b8999R60-R61).